pkg/controller/job/job_controller_test.go

```diff
@@ -6242,66 +6330,119 @@ func TestJobBackoffForOnFailure(t *testing.T) {
 		podPhase      v1.PodPhase
 
 		// expectations
-		expectedActive          int32
-		expectedSucceeded       int32
-		expectedFailed          int32
-		expectedCondition       *batch.JobConditionType
-		expectedConditionReason string
+		expectedActive     int32
+		expectedSucceeded  int32
+		expectedFailed     int32
+		expectedConditions []batch.JobCondition
 	}{
 		"backoffLimit 0 should have 1 pod active": {
 			1, 1, 0,
```

Comment:
while we are at it, please add the name of the fields to all the test cases here. This is impossible to read right now.

Reply:
Yeah, updated